packages/atomic/src/decorators/with-tailwind-styles.ts

```diff
@@ -1,30 +1,78 @@
-import {CSSResult, type CSSResultGroup, unsafeCSS} from 'lit';
+import {type CSSResult, type CSSResultGroup, unsafeCSS} from 'lit';
 import theme from '@/src/utils/coveo.tw.css';
 import styles from '@/src/utils/tailwind.global.tw.css';
 import utilities from '@/src/utils/tailwind-utilities/utilities.tw.css';
 
-export function withTailwindStyles<
+const tailwindPropertiesSheet: CSSStyleSheet | null =
+  typeof window !== 'undefined'
+    ? (() => {
+        const propCss = styles as unknown as string[] | string;
+        const fullCss = Array.isArray(propCss) ? propCss.join('') : propCss;
+
+        // Extract only the variable definitions inside @layer properties
+        const match = fullCss.match(
+          /@layer\s+properties\s*{[^{}]*@supports[^{}]*{([^}]*)}|@layer\s+properties\s*{([^}]*)}/
+        );
+        const cssString = (match?.[1] || match?.[2] || '').trim();
+        if (!cssString) {
+          return null;
+        }
+        const wrappedCss = `@layer properties { ${cssString} }`;
+        const sheet = new CSSStyleSheet();
+        sheet.replaceSync(wrappedCss);
+        return sheet;
+      })()
+    : null;
+
+// biome-ignore lint/suspicious/noExplicitAny: <>
+function injectTailwindProperties(element: any) {
+  if (typeof window === 'undefined' || !tailwindPropertiesSheet) return;
+
+  const parent = element.getRootNode();
+  const isDocumentOrShadowRoot =
```

Comment:
Should it be `isParentDocumentOrShadowRoot`?

Alternatively, you could extract this into a typeguard function, e.g.,

```typescript
function isDocumentOrShadowRoot(element: any): element is Document | ShadowRoot {
  return (
    element instanceof Document || element instanceof ShadowRoot
  );
}
```